src/Umbraco.Infrastructure/BackOffice/IdentityAuditEventArgs.cs

```diff
@@ -0,0 +1,132 @@
+        /// </summary>
+        public int AffectedUser { get; private set; }
+
+        /// <summary>
+        /// If a user is performing an action on a different user, then this will be set. Otherwise it will be -1
+        /// </summary>
+        public int PerformingUser { get; private set; }
+
+        /// <summary>
+        /// An optional comment about the action being logged
+        /// </summary>
+        public string Comment { get; private set; }
+
+        /// <summary>
+        /// This property is always empty except in the LoginFailed event for an unknown user trying to login
+        /// </summary>
+        public string Username { get; private set; }
+
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        /// <param name="action"></param>
+        /// <param name="ipAddress"></param>
+        /// <param name="comment"></param>
+        /// <param name="performingUser"></param>
+        /// <param name="affectedUser"></param>
+        public IdentityAuditEventArgs(AuditEvent action, string ipAddress, string comment = null, int performingUser = -1, int affectedUser = -1)
+        {
+            DateTimeUtc = DateTime.UtcNow;
+            Action = action;
+
+            IpAddress = ipAddress;
+            Comment = comment;
+            AffectedUser = affectedUser;
+
+            PerformingUser = performingUser == -1
+                ? GetCurrentRequestBackofficeUserId()
+                : performingUser;
+        }
+
+        /// <summary>
+        /// Creates an instance without a performing or affected user (the id will be set to -1)
+        /// </summary>
+        /// <param name="action"></param>
+        /// <param name="ipAddress"></param>
+        /// <param name="username"></param>
+        /// <param name="comment"></param>
+        public IdentityAuditEventArgs(AuditEvent action, string ipAddress, string username, string comment)
+        {
+            DateTimeUtc = DateTime.UtcNow;
+            Action = action;
+
+            IpAddress = ipAddress;
+            Username = username;
+            Comment = comment;
+
+            PerformingUser = -1;
+        }
+
+        public IdentityAuditEventArgs(AuditEvent action, string ipAddress, string username, string comment, int performingUser)
+        {
+            DateTimeUtc = DateTime.UtcNow;
+            Action = action;
+
+            IpAddress = ipAddress;
+            Username = username;
+            Comment = comment;
+
+            PerformingUser = performingUser == -1
+                ? GetCurrentRequestBackofficeUserId()
+                : performingUser;
+        }
+
+        /// <summary>
+        /// Returns the current logged in backoffice user's Id logging if there is one
+        /// </summary>
+        /// <returns></returns>
+        protected int GetCurrentRequestBackofficeUserId()
+        {
+            var userId = -1;
+            /*var backOfficeIdentity = Thread.CurrentPrincipal.GetUmbracoIdentity();
```

Comment:
Is this one of the TODOs?

Reply:
Uncommenting this code still builds so I'm wondering if it was just part of the refactoring process?

Reply:
Sorry, I missed this one when I reviewed my changes. It had to be removed since Thread.CurrentPrincipal is no longer supported in ASP.NET Core: https://docs.microsoft.com/en-us/aspnet/core/migration/claimsprincipal-current?view=aspnetcore-3.1. Again, this requires more Umbraco knowledge than I have to refactor this. I've added it to the notes in the PR write up.

Since both ASP.NET and ASP.NET Core are using this method, you could multi-target by wrapping it in an ifdef for now.

Reply:
thanks for the link, i'll have a look at this one 👍